test/gtest/ucp/test_ucp_tag_offload.cc

```diff
@@ -403,6 +404,48 @@ UCP_INSTANTIATE_TEST_CASE_TLS(test_ucp_tag_offload_multi, all_rcdc,
                               "\\rc,\\ud,rc_x,dc_x")
 
 
+class test_ucp_tag_offload_selection : public test_ucp_tag_offload {
+public:
+
+    void init()
+    {
+        m_env.push_back(new ucs::scoped_setenv("UCX_RC_TM_ENABLE", "y"));
+        test_ucp_tag::init();
+    }
+};
+
+UCS_TEST_P(test_ucp_tag_offload_selection, tag_lane)
+{
+    ucp_ep_h ep          = sender().ep();
+    bool has_tag_offload = false;
+    bool has_shm_or_self = false;
+
+    for (ucp_rsc_index_t idx = 0; idx < sender().ucph()->num_tls; ++idx) {
+        if (ucp_wireup_is_lane_self_or_shm(ep, idx)) {
+            has_shm_or_self = true;
+        }
+
+        uct_iface_attr_t *attr = ucp_worker_iface_get_attr(sender().worker(), idx);
+        if (attr->cap.flags & UCT_IFACE_FLAG_TAG_EAGER_BCOPY) {
+            // We do not have transports with partial tag offload support
+            EXPECT_TRUE(attr->cap.flags & UCT_IFACE_FLAG_TAG_RNDV_ZCOPY);
+            has_tag_offload = true;
+        }
+    }
+
+    if (ucp_ep_is_tag_offload_enabled(ucp_ep_config(ep))) {
+        // If shm or self transport exists it should be used for tag matching
```

Comment:
maybe form this code the other way around: if have_tag_offload && !have_shm_or_self -> expect tag offload to be enabled
also comment `// If shm or self transports exist they would be used for tag matching rather than network offload`

Reply:
I also check that tag offload is not used when we have shm lane, so it will be ~the same.
But ok, will change this way

Reply:
yes, i've meant to check both, but in reverse order (i think it will make more sense)